### PR TITLE
new auth/profile/wallet logic WIP [bugs]

### PR DESCRIPTION
**Pull request by jjjuk**

### CODE REVIEW

File: src/layout/index.tsx [hunk 1]
```diff
@@ -22,6 +25,17 @@ export default function Layout({ children }: PropsWithChildren<any>) {
     } else {
       dispatch(UiActions.setPage())
     }
+
+    getAccountDetails()
+      .then((user) => {
+        if (user) {
+          dispatch(AuthActions.fetchSuccess(user))
+          // TODO: server to client wallet connection logic
```
Comment (EvolveArt) [inline src/layout/index.tsx:33]: more details on that please

**Review by EvolveArt — CHANGES_REQUESTED** [review]

File: src/components/Pages/Auth/PassReset.tsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+  const onVisibilityClick: React.MouseEventHandler<HTMLDivElement> = () => {
+    setForm(
+      produce(form, (draft) => {
+        draft.visible = !draft.visible
+      })
+    )
+  }
+
+  const submit = () => {}
```
[the remark this answers is not in the document]
Comment (jjjuk) [reply src/components/Pages/Auth/PassReset.tsx]: fixed

**Review by EvolveArt — COMMENTED** [review]
Do we check that the user doesn't change to the same password than before ?

**Comment by jjjuk** [comment]
> Do we check that the user doesn't change to the same password than before ?
No, we don't

File: next.config.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ const nextConfig = {
   publicRuntimeConfig: {
     //   NEXT_PUBLIC_API_URL: 'https://zkpad-api.herokuapp.com/api/graphql',
     NEXT_PUBLIC_REST_API_URL:
-      'http://localhost:4004/api' /* 'https://zkpad-api.herokuapp.com/api' */,
+      'https://8ce1-146-255-233-146.eu.ngrok.io/api' /* 'https://zkpad-api.herokuapp.com/api' */,
```
Comment (EvolveArt) [inline next.config.js:9]: wrong url here ?

File: src/components/Pages/Auth/AuthForm.tsx [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+  const disabled =
+    !form.payload.email ||
+    !form.payload.password ||
+    form.errors.password ||
+    form.errors.email ||
+    (signUp && !form.agree)
+
+  return (
+    <React.Fragment>
+      <div className="mb-28px">
```
[the remark this answers is not in the document]
Comment (jjjuk) [reply src/components/Pages/Auth/AuthForm.tsx:140]: We haven't had any of these in the app, so... I prefer just use immer for forms. It's not even a "package" its just an immutability wrapper. Don't see it more difficult to work with, prefer not to grow packages size and have more control over process)

[the remark this answers is not in the document]
Comment (jjjuk) [reply src/components/Pages/Auth/AuthForm.tsx:140]: About 
```html 
<form>...</form>
``` 
it's a good point 🤣

File: src/hooks/useEagerConnect.js [hunk 5]
```diff
@@ -13,8 +13,11 @@ export default function useEagerConnect() {
     const _connector = Object.keys(SUPPORTED_WALLETS).find(
       (key) => _walletName === SUPPORTED_WALLETS[key].name
     )
+
+    if (!_connector) return // 🫶
```
[the remark this answers is not in the document]
Comment (jjjuk) [reply src/hooks/useEagerConnect.js:17]: Don't need to continue execution if there is no _connector)

File: src/layout/index.tsx (hunk 1 above)
Comment (jjjuk) [reply src/layout/index.tsx:33]: We discussed that in some scenario we need to connect wallet from data fetched from the server, so having just an address, but not from useStarknetReact() and extensions. You said that it's your job. So this comment tells the place where this logic should start. If it's inconvenient to you, feel free to suggest your options!

**Pull request merged**